### PR TITLE
Define valid calendar and time zone IDs

### PR DESCRIPTION
**Pull request by ptomato**
Closes: #665

### CODE REVIEW

File: docs/calendar.md [hunk 1]
```diff
@@ -41,6 +41,8 @@ For specialized applications where you need to do calculations in a calendar sys
 To do this, create a class inheriting from `Temporal.Calendar`, call `super()` in the constructor with a calendar identifier, and implement all the methods.
 Any subclass of `Temporal.Calendar` will be accepted in Temporal APIs where a built-in `Temporal.Calendar` would work.
 
+The identifier of a custom calendar must not be any of the built-in calendar identifiers, and must consist of one or more components of between 2 and 8 ASCII alphanumeric characters each, separated by dashes, as described in [BCP 47](https://tools.ietf.org/html/bcp47#section-2.1).
```
Comment (sffc) [inline docs/calendar.md:44]: Unicode extension value subtags are 3-8, not 2-8

https://unicode.org/reports/tr35/tr35.html#Unicode_locale_identifier

Comment (sffc) [inline docs/calendar.md:44]: The exact syntax is likely to depend on the outcome of the RFC 3339 discussions.

Comment (sffc) [inline docs/calendar.md:44]: Why is it explicitly illegal for a custom calendar to have the same ID as a built-in calendar?

Comment (ptomato) [reply docs/calendar.md:44]: Ah, so https://tools.ietf.org/html/bcp47#section-2.1 doesn't contain the full information? Will change this.

Comment (ptomato) [reply docs/calendar.md:44]: Now that I think about it, I guess we do want to allow this, for overriding the built-in ones.

Comment (ptomato) [reply docs/calendar.md:44]: I'll add an editorial note to that effect.

Comment (ptomato) [reply docs/calendar.md]: (Added in IsValidCalendarId in the spec)

**Review by sffc — APPROVED** [review]

File: docs/calendar.md (hunk 1 above)
Comment (sffc) [reply docs/calendar.md]: Yeah; in ECMA-402 we use the UTS 35 version of locale identifiers, which is a subset of IETF.

**Comment by ptomato** [comment]
Thanks for the review!

**Pull request merged**